arc.m

```diff
@@ -60,17 +87,13 @@ - (void)release;
 
 static inline struct arc_tls* getARCThreadData(void)
 {
-#ifdef NO_PTHREADS
-	return NULL;
-#else
-	struct arc_tls *tls = pthread_getspecific(ARCThreadKey);
+	struct arc_tls *tls = arc_tls_load(ARCThreadKey);
```

Comment:
We still need to keep the `NO_PTHREADS` case for platforms that both are not Windows and don't support pthreads.